File: tks/WebService/src/main/kotlin/webservice/adapters/UserWebServiceAdapter.java
package webservice.adapters;

import domain.exceptions.TypeValidationFailedException;
import domain.exceptions.UserNotFoundException;
import domain.model.values.Email;
import application.services.UserService;
import webservice.dto.UserSoapDto;
import webservice.mappers.IUserMapper;

import javax.inject.Inject;

public class UserWebServiceAdapter {

    @Inject
    private UserService userService;
    @Inject
    private IUserMapper mapper;

    public UserSoapDto getUser(String email) throws TypeValidationFailedException, UserNotFoundException {
        var key = new Email(email);
        var user = userService.getDetails(key);
        return mapper.toDto(user);
    }
}

File: tks/WebService/src/main/kotlin/webservice/adapters/UserWebServiceAdapter.kt
package webservice.adapters

import application.services.UserService
import domain.exceptions.TypeValidationFailedException
import domain.exceptions.UserNotFoundException
import domain.model.values.Email
import ports.primary.combined.IUserService
import webservice.dto.UserSoapDto
import webservice.mappers.UserMapper
import javax.inject.Inject

class UserWebServiceAdapter(
    @Inject private var userService: IUserService? = null,
    @Inject private var mapper: UserMapper? = null
) {


    @Throws(TypeValidationFailedException::class, UserNotFoundException::class)
    fun getUser(email: String): UserSoapDto {
        val key = Email(email)
        val user = userService!!.getDetails(key)
        return mapper!!.toDto(user)!!
    }
}